addons/utm/views/utm_campaign_views.xml

```diff
@@ -0,0 +1,164 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record model="ir.ui.view" id="view_utm_campaign_view_search">
+        <field name="name">utm.campaign.search</field>
+        <field name="model">utm.campaign</field>
+        <field name="arch" type="xml">
+            <search string="UTM Campaigns">
+                <field name="name" string="Campaigns"/>
+                <field name="tag_ids"/>
+                <field name="user_id"/>
+                <group expand="0" string="Group By">
+                    <filter string="Stage" name="group_stage_id"
+                        context="{'group_by': 'stage_id'}"/>
+                    <filter string="Responsible" name="group_user_id"
+                        context="{'group_by': 'user_id'}"/>
+                </group>
+            </search>
+        </field>
+    </record>
+
+    <record model="ir.ui.view" id="utm_campaign_view_form">
+        <field name="name">utm.campaign.form</field>
+        <field name="model">utm.campaign</field>
+        <field name="arch" type="xml">
+            <form string="UTM Campaign">
+                <sheet>
+                    <div class="oe_button_box d-flex justify-content-end" name="button_box">
+                    </div>
+                    <group id="top-group">
+                        <field name="name" string="Campaign Name"/>
+                        <field name="user_id"/>
+                        <field name="tag_ids" widget="many2many_tags" options="{'color_field': 'color', 'no_create_edit': True}"/>
+                    </group>
+                    <notebook>
+                    </notebook>
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+    <record model="ir.ui.view" id="utm_campaign_view_tree">
+        <field name="name">utm.campaign.tree</field>
```

Comment:
Guidelining notes: we could update some name, utm.campaign.view.tree/form/.. :) .